### PR TITLE
Add top script to run the different Rule Enforcer scripts

### PR DESCRIPTION
**Pull request by cedelavergne-ledger**
Add a top script gathering the different check commands in a single place.
This script can be called from the command line, locally or inside a docker.

This is the 1st step to be able to run the equivalent of the Guideline Enforcer workflow from VSCode extension.

This has been tested using the following apps triggering this branch (in a draft PR):
- app-boilerplate
- app-boilerplate-rust
- app-plugin-boilerplate
- app-ethereum
- app-flow

### CODE REVIEW

File: .github/workflows/_check_clang_static_analyzer.yml [hunk 1]
```diff
@@ -29,24 +38,32 @@ jobs:
...
           submodules: recursive
 
       - name: Run Clippy
         if: ${{ inputs.is_rust == 'true'}}
         run: |
-          cd ${{ inputs.relative_app_directory }} && \
-          DEVICE_NAME="$(echo ${{ matrix.device }} | sed 's/nanosp/nanosplus/')" && \
-          cargo +$RUST_NIGHTLY clippy --target ${DEVICE_NAME} -- -Dwarnings
+            ./ledger-app-workflows/scripts/check_all.sh -r -c clang -t ${{ matrix.device }} \
+              -a ./app-repository -b ${{ inputs.relative_app_directory }}
```
Comment (tdejoigny-ledger) [inline .github/workflows/_check_clang_static_analyzer.yml:59]: `clang ` arg seems not to be the best wording for rust app

File: scripts/check_all.sh [hunk 2]
```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env bash
```
Comment (tdejoigny-ledger) [inline scripts/check_all.sh:1]: do we have the readme check here ?

File: .github/workflows/_check_clang_static_analyzer.yml (hunk 1 above)
Comment (cedelavergne-ledger) [reply .github/workflows/_check_clang_static_analyzer.yml:59]: Sure. Fixed!

File: scripts/check_all.sh (hunk 2 above)
Comment (cedelavergne-ledger) [reply scripts/check_all.sh:1]: Yes, it is included (check line 182)

**Review by fbeutin-ledger — APPROVED** [review]

**Pull request merged**